src/Common/src/Interop/Unix/System.Security.Cryptography.Native/Interop.ASN1.cs

```diff
@@ -0,0 +1,112 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Diagnostics;
+using System.Runtime.InteropServices;
+using System.Security.Cryptography;
+using System.Text;
+
+using Microsoft.Win32.SafeHandles;
+
+internal static partial class Interop
+{
+    internal static partial class Crypto
+    {
+        internal const int NID_undef = 0;
+
+        internal const int NID_secp224r1 = 713; // NIST-224
+        internal const int NID_secp384r1 = 715; // NIST-384
+        internal const int NID_secp521r1 = 716; // NIST-521
+
+        [DllImport(Libraries.CryptoNative)]
+        internal static extern long Asn1IntegerGet(IntPtr a);
+
+        [DllImport(Libraries.CryptoNative, CharSet = CharSet.Ansi)]
+        internal static extern SafeAsn1ObjectHandle ObjTxt2Obj(string s);
+
+        [DllImport(Libraries.CryptoNative, CharSet = CharSet.Ansi)]
+        private static extern int ObjObj2Txt([Out] StringBuilder buf, int buf_len, IntPtr a);
+
+        [DllImport(Libraries.CryptoNative, CharSet = CharSet.Ansi)]
+        internal static extern IntPtr GetFriendlyNameSharedObject(string friendlyName);
+
+        [DllImport(Libraries.CryptoNative, CharSet = CharSet.Ansi)]
+        internal static extern int ObjSn2Nid(string sn);
+
+        // Returns shared pointers, should not be tracked as a SafeHandle.
+        [DllImport(Libraries.CryptoNative)]
+        internal static extern IntPtr ObjNid2Obj(int nid);
+        
+        [DllImport(Libraries.CryptoNative)]
+        internal static extern void Asn1ObjectFree(IntPtr o);
+
+        [DllImport(Libraries.CryptoNative)]
+        internal static unsafe extern SafeAsn1BitStringHandle D2IAsn1BitString(IntPtr zero, byte** ppin, int len);
+
+        [DllImport(Libraries.CryptoNative)]
+        internal static extern void Asn1BitStringFree(IntPtr o);
+
+        [DllImport(Libraries.CryptoNative)]
+        internal static unsafe extern SafeAsn1OctetStringHandle D2IAsn1OctetString(IntPtr zero, byte** ppin, int len);
+
+        [DllImport(Libraries.CryptoNative)]
+        internal static extern SafeAsn1OctetStringHandle Asn1OctetStringNew();
+
+        [DllImport(Libraries.CryptoNative)]
+        [return: MarshalAs(UnmanagedType.Bool)]
+        internal static extern bool Asn1OctetStringSet(SafeAsn1OctetStringHandle o, byte[] d, int len);
+
+        [DllImport(Libraries.CryptoNative)]
+        internal static extern void Asn1OctetStringFree(IntPtr o);
+
+        [DllImport(Libraries.CryptoNative)]
+        internal static extern void Asn1StringFree(IntPtr o);
+
+        internal static string OBJ_obj2txt_helper(IntPtr asn1ObjectPtr)
```

Comment:
I did a terrible job of naming this; and since it's not directly calling OBJ_obj2txt anymore, maybe the name should be spruced up.

GetOidValue, perhaps?